### PR TITLE
[lntest] remove rpctest build flag

### PR DESCRIPTION
**Pull request by halseth**
Now that the itests are in a separate package, we can remove the rpctest
build flag. This makes code completion and syntax checking work since
the files are no longer "hidden" behind the flag.

To make the tools able to properly compile the whole package without any
build flag, we also change the btcd backend compile when !neutrino is
active, instead of requiring the btcd flag.

### CODE REVIEW

File: lntest/btcd.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// +build btcd
+// +build !neutrino
```
Comment (wpaulino) [inline lntest/btcd.go]: I assume we can set multiple of these once we add bitcoind as well?

File: make/testing_flags.mk [hunk 2]
```diff
@@ -54,15 +54,12 @@ UNIT := $(GOLIST) | $(XARGS) env $(GOTEST) -tags="$(DEV_TAGS) $(LOG_TAGS)" $(TES
...
-# Construct the integration test command with the added build flags.
-ITEST_TAGS := $(DEV_TAGS) rpctest chainrpc walletrpc signrpc invoicesrpc autopilotrpc routerrpc
-
 # Default to btcd backend if not set.
-ifneq ($(backend),)
-ITEST_TAGS += ${backend}
-else
-ITEST_TAGS += btcd
+ifeq ($(backend),)
+backend = btcd
```
Comment (wpaulino) [inline make/testing_flags.mk]: Prior commit removes the btcd tag so this seems no longer needed?

File: make/testing_flags.mk [hunk 3]
```diff
@@ -54,15 +54,12 @@ UNIT := $(GOLIST) | $(XARGS) env $(GOTEST) -tags="$(DEV_TAGS) $(LOG_TAGS)" $(TES
...
 endif
 
+# Construct the integration test command with the added build flags.
+ITEST_TAGS := $(DEV_TAGS) rpctest chainrpc walletrpc signrpc invoicesrpc autopilotrpc routerrpc $(backend)
```
Comment (cfromknecht) [inline make/testing_flags.mk]: still passing `rpctest` tag?

File: make/testing_flags.mk (hunk 2 above)
Comment (halseth) [reply make/testing_flags.mk]: It is only needed for the `running integration tests with <backend> chain backend` logs

File: make/testing_flags.mk (hunk 3 above)
Comment (halseth) [reply make/testing_flags.mk]: Still needed here: https://github.com/lightningnetwork/lnd/blob/master/sweep/defaults_rpctest.go#L1

**Comment by cfromknecht** [comment]
needs rebase

**Comment by halseth** [comment]
Rebased

**Comment by cfromknecht** [comment]
needs another!

**Comment by Roasbeef** [comment]
> needs another!

another1

**Comment by halseth** [comment]
rebased2

**Comment by halseth** [comment]
Added to commits to address linter errors that now show up because of the build flag removal.

**Comment by cfromknecht** [comment]
```
==================
WARNING: DATA RACE
Read at 0x0000020e6bd0 by goroutine 25:
  github.com/lightningnetwork/lnd/lntest.newNode()
      /home/travis/gopath/src/github.com/lightningnetwork/lnd/lntest/node.go:84 +0x99d
  github.com/lightningnetwork/lnd/lntest.(*NetworkHarness).newNode()
      /home/travis/gopath/src/github.com/lightningnetwork/lnd/lntest/harness.go:344 +0x186
  github.com/lightningnetwork/lnd/lntest.(*NetworkHarness).SetUp.func1()
      /home/travis/gopath/src/github.com/lightningnetwork/lnd/lntest/harness.go:251 +0xd5
Previous write at 0x0000020e6bd0 by goroutine 26:
  github.com/lightningnetwork/lnd/lntest.newNode()
      /home/travis/gopath/src/github.com/lightningnetwork/lnd/lntest/node.go:286 +0xaf3
  github.com/lightningnetwork/lnd/lntest.(*NetworkHarness).newNode()
      /home/travis/gopath/src/github.com/lightningnetwork/lnd/lntest/harness.go:344 +0x186
  github.com/lightningnetwork/lnd/lntest.(*NetworkHarness).SetUp.func2()
      /home/travis/gopath/src/github.com/lightningnetwork/lnd/lntest/harness.go:251 +0xd5
Goroutine 25 (running) created at:
  github.com/lightningnetwork/lnd/lntest.(*NetworkHarness).SetUp()
      /home/travis/gopath/src/github.com/lightningnetwork/lnd/lntest/harness.go:133 +0x1d7
  github.com/lightningnetwork/lnd/lntest/itest.TestLightningNetworkDaemon()
      /home/travis/gopath/src/github.com/lightningnetwork/lnd/lntest/itest/lnd_test.go:14368 +0x8d1
  testing.tRunner()
      /home/travis/.gimme/versions/go1.12.6.linux.amd64/src/testing/testing.go:865 +0x163
Goroutine 26 (running) created at:
  github.com/lightningnetwork/lnd/lntest.(*NetworkHarness).SetUp()
      /home/travis/gopath/src/github.com/lightningnetwork/lnd/lntest/harness.go:142 +0x23d
  github.com/lightningnetwork/lnd/lntest/itest.TestLightningNetworkDaemon()
      /home/travis/gopath/src/github.com/lightningnetwork/lnd/lntest/itest/lnd_test.go:14368 +0x8d1
  testing.tRunner()
      /home/travis/.gimme/versions/go1.12.6.linux.amd64/src/testing/testing.go:865 +0x163
```
travis picked up a race condition

**Comment by halseth** [comment]
> travis picked up a race condition

Ugh, unit tests will attempt to run the itests now, since they are no longer behind the build flag. Maybe this wasn't such a great idea.

**Comment by halseth** [comment]
Closing in favor of https://github.com/lightningnetwork/lnd/pull/3382

**Pull request closed without merging**